src/sentry/api/serializers/models/group.py

```diff
@@ -347,8 +349,20 @@ def serialize(self, obj, attrs, user):
         # do not return the permalink which contains private information i.e. org name.
         request = env.request
         is_superuser = request and is_active_superuser(request) and request.user == user
-        if is_superuser or (
-            user.is_authenticated() and user.get_orgs().filter(id=obj.organization.id).exists()
+
+        # If user is a sentry_app then it's a proxy user meaning we can't do a org lookup via `get_orgs()`
+        # because the user isn't an org member. Instead we can use the auth token and the installation
+        # it's associated with to find out what organization the token has access to.
+        is_valid_sentryapp = False
+        if hasattr(request.user, "is_sentry_app") and isinstance(request.auth, ApiToken):
+            is_valid_sentryapp = SentryAppInstallationToken.has_organization_access(
+                request.auth, obj.organization
+            )
+
+        if (
+            is_superuser
+            or (user.is_authenticated() and user.get_orgs().filter(id=obj.organization.id).exists())
```

Comment:
@MeredithAnya nit: maybe we should assign this to a variable like `is_authenticated_user_for_org` to make this cleaner.

Reply:
nvm, it would get awkward b/c we only need to do the check sometimes. Ignore this :)